pkg/instance/handler.go

```diff
@@ -156,7 +164,13 @@ func (h Handler) Deploy(c *gin.Context) {
 	}
 
 	if deploy {
-		err = h.instanceService.Deploy(token, savedInstance)
+		decryptedInstance, err := h.instanceService.FindByIdDecrypted(i.ID)
```

Comment:
It looks a bit repetitive having to do the saving, findBy multiple times in this handler.

Did we not already do that above in line 138 https://github.com/dhis2-sre/im-manager/pull/121/files#diff-0a77e42443c6231c5241d66235af8a4e3b08dfe25a653f17991b4b02922df648R138 🤔 

Also https://github.com/dhis2-sre/im-manager/blob/20f3ff12f5d743b1015b04dfa552977ca8adde7b/pkg/instance/service.go#L182-L188 is doing a `s.instanceRepository.Save(instance)`

I wonder if we can somehow reduce the number of calls we make to the DB.

Reply:
I just saw the changes to the `instanceService` and that you removed these lines. Can you please explain why you removed them?